pkg/promclient/promclient.go

```diff
@@ -376,6 +379,32 @@ func PromqlQueryInstant(ctx context.Context, logger log.Logger, base *url.URL, q
 	return vec, warnings, nil
 }
 
+// GetPromVersion will return the version of Prometheus by querying /version Prometheus endpoint.
+func PromVersion(logger log.Logger, base *url.URL) (*version.Version, error) {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+
+	u := *base
+	u.Path = path.Join(u.Path, "/version")
+	resp, err := http.Get(u.String())
+	if err != nil {
+		return nil, errors.Wrapf(err, "request version against %s", u.String())
+	}
+	defer runutil.CloseWithLogOnErr(logger, resp.Body, "query body")
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Errorf("failed to read body")
```

Comment:
Very minor thing: `errors.Errorf` is not needed here since you aren't doing any formatting. Just use `errors.New`.

Reply:
Thx!